### ISSUE TITLE
Will bs4Dash going to support bootstrap 5?

### ISSUE THREAD

**Issue by agenius-mohammed-ali** [RinteRface/bs4Dash#362]
Hi,
Thank you for this great package. Only I wonder if there are plans to upgrade or create a version that supports bootstrap 5 and next versions of bootstraps?
Thank you

**Comment by DivadNojnarg** [comment]
Hi @agenius-mohammed-ali ,

Thanks for your question! At the moment, here are my thoughts:

- Going to Bootstrap 5 -> This is what was supposed to happen earlier this year. However, the underlying template AdminLTE 4 is not yet officially released with Bootstrap 5 so I can't start before it is stable. Besides, in addition to bring nothing really outstanding (appart from few color updates 😞) it breaks many existing features like the right sidebar, which would require time to be re-included. 

- Currently, I think you can pretty much achieve the same result with `{bslib}`. I also believe, the sidebar navigation will be included soon (poke @cpsievert).

**Comment by agenius-mohammed-ali** [comment]
Thanks a lot @DivadNojnarg for your answer. I saw, and use, `fresh` package with `bs4Dash` are there any tutorial that we can refer to with `bslib`, though I still prefer working with `fresh` package. Thank you

**Comment by cpsievert** [comment]
@agenius-mohammed-ali this, currently, is the best "getting started with `{bslib}` dashboards" article https://rstudio.github.io/bslib/articles/dashboards/index.html

**Comment by agenius-mohammed-ali** [comment]
Thanks @cpsievert so much for sharing the article.  I am impressed with how far `bslib` package achieved recently. Only I have the following questions:

- The package lifecycle is still "experimental", will it be stable anytime soon?
- The package creates dashboards from scratch, or in other words, it is a replacement for bs4Dash and shiny dashboard, is that right? (cc. @DivadNojnarg )
Thank you

**Comment by agenius-mohammed-ali** [comment]
Thanks a lot @cpsievert 
@cb4ds

**Comment by AshesITR** [comment]
According to [AdminLTEs GitHub](https://github.com/ColorlibHQ/AdminLTE) Version 3.2.0 should support bootstrap 5.
Is that release sufficiently stable?

**Comment by DivadNojnarg** [comment]
See above 

> Going to Bootstrap 5 -> This is what was supposed to happen earlier this year. However, the underlying template AdminLTE 4 is not yet officially released with Bootstrap 5 so I can't start before it is stable. Besides, in addition to bring nothing really outstanding (appart from few color updates 😞) it breaks many existing features like the right sidebar, which would require time to be re-included.

**Comment by AshesITR** [comment]
So you're waiting for AdminLTE 4.0?
Please let me know when you start development on bs5 support, I'm willing to help.

**Comment by jimbrig** [comment]
@DivadNojnarg @AshesITR @cpsievert etc., the latest AdminLTE Beta Release: [v4.0.0-beta1](https://github.com/ColorlibHQ/AdminLTE/releases/tag/v4.0.0-beta1) supports bootstrap 5 just FYI. Seems like the release is pretty stable but there is still WIP mostly related to docs, examples, etc. The GitHub Project Board is here: https://github.com/orgs/ColorlibHQ/projects/1

**Comment by DivadNojnarg** [comment]
@jimbrig

The beta version does not seem super polished:

- My main concern: there is no documentation about the new JS API. I don't know how much re-work and compatibility is has with the old API. The old doc was more comprehensive: https://adminlte.io/docs/3.1/. Looking at the new code this does not seem much different: https://github.com/ColorlibHQ/AdminLTE/blob/master/src/ts/card-widget.ts but maybe there are edge cases.
- The right sidebar is still missing. This could be done with an offCanvas but that's more work.
- I see some glitches (see video): shaking element.
- Probably other hidden things.

I can have a naive look within bs4Dash and see if there are other outstanding issues.

https://github.com/user-attachments/assets/a4a92383-4d51-4078-91f2-5656bbe006ea

**Comment by DivadNojnarg** [comment]
AdminLTE Beta 2 released: https://github.com/ColorlibHQ/AdminLTE/releases/tag/v4.0.0-beta2. The blinking seems to be fixed.

**Comment by DivadNojnarg** [comment]
@jimbrig I started to look at it closely: https://github.com/RinteRface/bs4Dash/commit/94626c6e09d0d8af02a3a9670b07b79a4b554457. However, it's much worse than I thought. There are so many missing pieces: socialBox, userBox, boxProfile, ... are just gone. I have no control over if this will be re-included at a later point and when. My suggestion would be a bs5Dash. I see no viable alternative to this, as it would be extremely frustrating for users to migrate and have so many breaking changes.

**Comment by jimbrig** [comment]
> @jimbrig I started to look at it closely: https://github.com/RinteRface/bs4Dash/commit/94626c6e09d0d8af02a3a9670b07b79a4b554457. However, it's much worse than I thought. There are so many missing pieces: socialBox, userBox, boxProfile, ... are just gone. I have no control over if this will be re-included at a later point and when. My suggestion would be a bs5Dash. I see no viable alternative to this, as it would be extremely frustrating for users to migrate and have so many breaking changes.

I agree with this all together and think that there are other benefits to this approach also. Always nice to start with a fresh package! 

If you guys do end up going this route I'd think of it as the bslib's shinydashboard.

**Comment by drewn82** [comment]
Is there any update on this?  Will the team keep looking to upgrade to the latest and greatest version?
